pkg/cmd/cli/common.go

```diff
@@ -18,11 +18,65 @@ package cli
 
 import (
 	"bufio"
+	"errors"
 	"fmt"
 	"os"
 	"strings"
+
+	"github.com/heptio/ark/pkg/client"
+	"github.com/heptio/ark/pkg/cmd/util/flag"
+	clientset "github.com/heptio/ark/pkg/generated/clientset/versioned"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 )
 
+// DeleteOptions contains parameters used for deleting a restore.
+type DeleteOptions struct {
+	Names     []string
+	all       bool
+	Selector  flag.LabelSelector
+	Confirm   bool
+	Client    clientset.Interface
+	Namespace string
+}
+
+// Complete fills in the correct values for all the options.
+func (o *DeleteOptions) Complete(f client.Factory, args []string) error {
+	o.Namespace = f.Namespace()
+	client, err := f.Client()
+	if err != nil {
+		return err
+	}
+	o.Client = client
+	o.Names = args
+	return nil
+}
+
+// Validate validates the fields of the DeleteOptions struct.
+func (o *DeleteOptions) Validate(c *cobra.Command, f client.Factory, args []string) error {
+	if o.Client == nil {
+		return errors.New("Ark client is not set; unable to proceed")
+	}
+	var (
+		hasNames    = len(o.Names) > 0
+		hasAll      = o.all
+		hasSelector = o.Selector.LabelSelector != nil
+	)
+	if !xor(hasNames, hasAll, hasSelector) {
+		return errors.New("you must specify exactly one of: specific restore name(s), the --all flag, or the --selector flag")
```

Comment:
This error currently hardcodes the string "restore" which isn't what we want since we're using this struct in multiple places.  I'd add a `NewDeleteOptions(singularTypeName string)` constructor function that takes the type this is being used for as an arg (`backup`, `restore`, `schedule`), then use that in this error message and in the `BindFlags` func.